gazebo/common/SphericalCoordinates.cc

```diff
@@ -246,6 +288,57 @@ double SphericalCoordinates::Distance(const ignition::math::Angle &_latA,
   return d;
 }
 
+//////////////////////////////////////////////////
+/// Based on Haversine formula (http://en.wikipedia.org/wiki/Haversine_formula).
+/// This takes into account the surface type.
+double SphericalCoordinates::DistanceBetweenPoints(
+    const ignition::math::Angle &_latA,
+    const ignition::math::Angle &_lonA,
+    const ignition::math::Angle &_latB,
+    const ignition::math::Angle &_lonB)
+{
+  ignition::math::Angle dLat = _latB - _latA;
+  ignition::math::Angle dLon = _lonB - _lonA;
+
+  double a = sin(dLat.Radian() / 2) * sin(dLat.Radian() / 2) +
+             sin(dLon.Radian() / 2) * sin(dLon.Radian() / 2) *
+             cos(_latA.Radian()) * cos(_latB.Radian());
+
+  double c = 2 * atan2(sqrt(a), sqrt(1 - a));
+  if (this->dataPtr->surfaceType == MOON_SCS)
+    return g_MoonRadius * c;
+  else
+    return g_EarthRadius * c;
```

Comment:
nit: I think you could just replace this if/else block with `return this->SurfaceRadius() * c;`, since that method also has the if/else block

Reply:
Done https://github.com/osrf/gazebo/pull/3250/commits/27936874696c7335fac1e6c58f48a6eff704fb8c